common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -27,16 +27,51 @@ public static String unquote(String text, String mark) {
 
   /**
    * Unquote Identifier which has " or ' or ` as mark.
+   * Replace doubled quotes within the string with single version
    * @param text string
    * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
    *     removed
    */
   public static String unquoteText(String text) {
-    if (isQuoted(text, "\"") || isQuoted(text, "'") || isQuoted(text, "`")) {
-      return text.substring(1, text.length() - 1);
+
+    if (text.length() < 2) {
+      return text;
+    }
+
+    StringBuilder textSB = new StringBuilder();
```

Comment:
move down to line ~60, since we don't need the stringbuilder until then.

Reply:
Can't do that, this covers the case where an empty string is passed. If an empty string is passed it'll error for index out of range because the setting of final character.

Reply:
I don't understand. 
`textSB` is not used until later in the method.  You don't need it declared until line 64.